lib/moonshot/json_stack_template.rb

```diff
@@ -0,0 +1,26 @@
+require 'json'
+require_relative 'stack_template'
+
+module Moonshot
+  # Handles JSON formatted AWS template files.
+  class JsonStackTemplate < StackTemplate
+    def initialize(filename, log:)
+      super(filename, log: log)
+      @body = JSON.parse(File.read(@filename)) if exists?
+    end
+
+    def parameters
+      @body.fetch('Parameters', {}).map do |k, v|
+        Parameter.new(k, v['Default'])
+      end
+    end
+
+    def resource_names
+      @body.fetch('Resources', {}).keys
+    end
+
+    def body
+      @body.to_json
```

Comment:
Introduced this to reduce unneccessary parsing/dumping to string/JSON (or YAML) and vice versa.